### PR TITLE
Replace d3-voronoi with d3-delaunay. 

### LINKED ISSUE #1492: Replace d3 voronoi with d3 delaunay
It's faster and numerically more stable.

https://twitter.com/mbostock/status/976537438121205760

### PR DESCRIPTION
**Pull request by domoritz**
Fixes #1492

### CODE REVIEW

**Comment by jheer** [comment]
Thanks for the PR! The failing tests are due to the test scene for the airports example, where the voronoi polygon path strings have changed.

I don't want to merge until we have a solution in place for d3/d3-delaunay#19. Vega can certainly encounter voronoi instances with less than 3 input points.

**Comment by domoritz** [comment]
Yeah, I agree that https://github.com/d3/d3-delaunay/issues/19 is a blocker for us. I wonder whether @mourner or @mbostock have this issue on their queue.

**Comment by domoritz** [comment]
There is also https://github.com/d3/d3-delaunay/issues/20 but it's less of an issue.

**Comment by jheer** [comment]
Hmm, I think d3/d3-delaunay#20 is absolutely a blocker as well. I've seen Vega users generate collinear points for a Voronoi diagram to enable a form of "nearest" selection.

**Comment by domoritz** [comment]
d3/d3-delaunay#20 has been fixed

**Comment by domoritz** [comment]
@jheer I updated the PR and fixed any remaining issues. Can you rebuild the airport example? I tried it locally and tests are passing but the results differ from what's happening on Travis.

**Comment by domoritz** [comment]
Ahh, it's because of #1893

File: packages/vega-voronoi/src/Voronoi.js [hunk 1]
```diff
@@ -22,25 +22,23 @@ Voronoi.Definition = {
...
+      delaunay, extent, voronoi, polygon, i, n;
 
   // configure and construct voronoi diagram
-  diagram = voronoi().x(_.x).y(_.y);
-  if (_.size) diagram.size(_.size);
-  else diagram.extent(_.extent || defaultExtent);
-
-  this.value = (diagram = diagram(data));
+  delaunay = Delaunay.from(data, _.x, _.y);
+  extent = _.size ? [0, 0 , ..._.size] : _.extent ? [..._.extent[0], _.extent[1][0]] : defaultExtent;
```
Comment (mbostock) [inline packages/vega-voronoi/src/Voronoi.js]: I think you’ll want `[..._.extent[0], ..._.extent[1]]` here?

Comment (domoritz) [reply packages/vega-voronoi/src/Voronoi.js]: Yes. Not sure what my past me was thinking here.

**Comment by domoritz** [comment]
I updated to the latest version of d3-delaunay and fixed the build. This is ready to go (squashed).

**Comment by jheer** [comment]
Why did you merge? I am working on this... Ugh.

**Comment by domoritz** [comment]
Ups, I took the approval as a permission to merge. I squashed all of the changes into one commit so hopefully it's not too confusing.

**Pull request merged**